### PR TITLE
Foreign Key Discovery

### PR DESCRIPTION
**Pull request by axadn**
### Description

-  fixes [https://github.com/strongloop/loopback-connector-mssql/issues/199](https://github.com/strongloop/loopback-connector-mssql/issues/199)
 - adds unit tests
### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by axadn** [comment]
I'm not sure why the "Commit Linter" check is failing. When I run 'npm lint' inside the repository folder it doesn't report any problems. Is the "Commit Linter" check running a different script?

**Comment by dhmlau** [comment]
Below is the commit linter error: 
```
**************************************************
**
**  Linting commit logs
**
**  3 problems found:
**    8ed0506 - add unit tests to discover.test.js. Test for correc: First line should be 50 characters or less (saw 120)
**    a3bc230 - fix buildQueryExportedForeignKeys function in lib/d: First line should be 50 characters or less (saw 104)
**    ac2bfcc - Fix buildForeignKeys method in discovery.js. Use th: First line should be 50 characters or less (saw 159)
**
**************************************************
```

There is a limit of the first line being restricted to 50 characters.  Please also refer to this [commit message guideline](https://loopback.io/doc/en/contrib/git-commit-messages.html#commit-messages). Thanks.

**Comment by axadn** [comment]
Thank you. I edited the commit messages.

**Comment by dhmlau** [comment]
@slnode test please

**Comment by dhmlau** [comment]
@slnode test please
there's timeout in one of the failed test cases.

**Comment by dhmlau** [comment]
@strongloop/sq-lb-apex , could you please review this PR? Thanks!

File: lib/discovery.js [hunk 1]
```diff
@@ -242,53 +220,24 @@ function mixinDiscovery(MsSQL) {
...
-   WHERE CONSTRAINT_TYPE = 'FOREIGN KEY' AND tc.TABLE_NAME='mytable';
-
    */
   MsSQL.prototype.buildQueryForeignKeys = function(owner, table) {
     var sql =
       'SELECT tc.TABLE_SCHEMA AS "fkOwner", tc.CONSTRAINT_NAME AS "fkName", tc.TABLE_NAME AS "fkTableName",' +
       ' kcu.COLUMN_NAME AS "fkColumnName", kcu.ORDINAL_POSITION AS "keySeq",' +
-      ' ccu.TABLE_SCHEMA AS "pkOwner", \'PK\' AS "pkName", ' +
-      ' ccu.TABLE_NAME AS "pkTableName", ccu.COLUMN_NAME AS "pkColumnName"' +
+      ' kcu2.TABLE_SCHEMA AS "pkOwner", rc.UNIQUE_CONSTRAINT_NAME AS "pkName",' +
```
Comment (raymondfeng) [inline lib/discovery.js]: I don't see the `kcu2` alias defined in the SQL, such as `as kcu2`.

Comment (axadn) [reply lib/discovery.js]: That alias comes from the join on line 239.

**Pull request closed without merging**